### ISSUE TITLE
1.9.7: test suite is failing in `061_Únicod€_file` unit

### ISSUE THREAD

**Issue by kloczek** [doxygen/doxygen#10053]
cmake settings (`cmake -L` output):
<details>

```console
-- Cache values
CMAKE_BUILD_TYPE:STRING=RelWithDebInfo
CMAKE_INSTALL_PREFIX:PATH=/usr
DOT:FILEPATH=/usr/bin/dot
Qt5Core_DIR:PATH=/usr/lib64/cmake/Qt5Core
Qt5Gui_DIR:PATH=/usr/lib64/cmake/Qt5Gui
Qt5Widgets_DIR:PATH=/usr/lib64/cmake/Qt5Widgets
Qt5Xml_DIR:PATH=/usr/lib64/cmake/Qt5Xml
Qt5_DIR:PATH=/usr/lib64/cmake/Qt5
SANITIZE_ADDRESS:BOOL=OFF
SANITIZE_LINK_STATIC:BOOL=OFF
SANITIZE_MEMORY:BOOL=OFF
SANITIZE_THREAD:BOOL=OFF
SANITIZE_UNDEFINED:BOOL=OFF
SPDLOG_BUILD_ALL:BOOL=OFF
SPDLOG_BUILD_BENCH:BOOL=OFF
SPDLOG_BUILD_EXAMPLE:BOOL=OFF
SPDLOG_BUILD_EXAMPLE_HO:BOOL=OFF
SPDLOG_BUILD_PIC:BOOL=OFF
SPDLOG_BUILD_SHARED:BOOL=OFF
SPDLOG_BUILD_TESTS:BOOL=OFF
SPDLOG_BUILD_TESTS_HO:BOOL=OFF
SPDLOG_BUILD_WARNINGS:BOOL=OFF
SPDLOG_CLOCK_COARSE:BOOL=OFF
SPDLOG_DISABLE_DEFAULT_LOGGER:BOOL=OFF
SPDLOG_ENABLE_PCH:BOOL=OFF
SPDLOG_FMT_EXTERNAL:BOOL=OFF
SPDLOG_FMT_EXTERNAL_HO:BOOL=OFF
SPDLOG_INSTALL:BOOL=OFF
SPDLOG_NO_ATOMIC_LEVELS:BOOL=OFF
SPDLOG_NO_EXCEPTIONS:BOOL=OFF
SPDLOG_NO_THREAD_ID:BOOL=OFF
SPDLOG_NO_TLS:BOOL=OFF
SPDLOG_PREVENT_CHILD_FD:BOOL=OFF
SPDLOG_SANITIZE_ADDRESS:BOOL=OFF
SPDLOG_TIDY:BOOL=OFF
SPDLOG_USE_STD_FORMAT:BOOL=OFF
SPDLOG_WCHAR_FILENAMES:BOOL=OFF
SPDLOG_WCHAR_SUPPORT:BOOL=OFF
build_app:BOOL=OFF
build_doc:BOOL=OFF
build_doc_chm:BOOL=OFF
build_parse:BOOL=OFF
build_search:BOOL=OFF
build_wizard:BOOL=ON
build_xmlparser:BOOL=ON
enable_console:BOOL=OFF
enable_coverage:BOOL=OFF
enable_lex_debug:BOOL=OFF
enable_tracing:BOOL=OFF
static_libclang:BOOL=OFF
use_libclang:BOOL=OFF
use_sqlite3:BOOL=OFF
win_static:BOOL=OFF
```
</details>

And test suite is failing in one unit
<details>

```console
+ /usr/bin/ctest --test-dir x86_64-redhat-linux-gnu --output-on-failure --force-new-ctest-process -j48 ' ' -E '^(012_cite)'
Internal ctest changing into directory: /home/tkloczko/rpmbuild/BUILD/doxygen-1.9.7/x86_64-redhat-linux-gnu
Test project /home/tkloczko/rpmbuild/BUILD/doxygen-1.9.7/x86_64-redhat-linux-gnu
      Start  1: 001_a
      Start  2: 002_addindex
      Start  3: 003_anchor
      Start  4: 004_arg
      Start  5: 005_attention
      Start  6: 006_author
      Start  7: 007_b
      Start  8: 008_brief
      Start  9: 009_bug
      Start 10: 010_c
      Start 11: 011_category
      Start 12: 013_class
      Start 13: 014_code
      Start 14: 015_cond
      Start 15: 016_copydoc
      Start 16: 017_copyright
      Start 17: 018_def
      Start 18: 019_defgroup
      Start 19: 020_only
      Start 20: 021_dontinclude
      Start 21: 022_dot
      Start 22: 023_e
      Start 23: 024_if
      Start 24: 025_example
      Start 25: 026_exception
      Start 26: 027_extends
      Start 27: 028_formula
      Start 28: 029_hideinit
      Start 29: 030_htmlinclude
      Start 30: 031_image
      Start 31: 032_include
      Start 32: 033_internal
      Start 33: 034_internal
      Start 34: 035_invariant
      Start 35: 036_link
      Start 36: 037_msc
      Start 37: 038_n
      Start 38: 039_name
      Start 39: 040_namespace
      Start 40: 041_overload
      Start 41: 042_package
      Start 42: 043_page
      Start 43: 044_section
      Start 44: 045_refitem
      Start 45: 046_related
      Start 46: 047_return
      Start 47: 048_showinit
      Start 48: 049_snippet
 1/98 Test  #3: 003_anchor ...................................   Passed    0.11 sec
      Start 49: 050_verbatim
 2/98 Test  #1: 001_a ........................................   Passed    0.12 sec
      Start 50: 051_escape
 3/98 Test  #2: 002_addindex .................................   Passed    0.12 sec
      Start 51: 052_tilde
 4/98 Test  #5: 005_attention ................................   Passed    0.11 sec
      Start 52: 053_tilde
 5/98 Test  #6: 006_author ...................................   Passed    0.12 sec
      Start 53: 054_parblock
 6/98 Test  #4: 004_arg ......................................   Passed    0.12 sec
      Start 54: 055_markdown
 7/98 Test #13: 014_code .....................................   Passed    0.11 sec
      Start 55: 056_latexinclude
 8/98 Test #10: 010_c ........................................   Passed    0.12 sec
      Start 56: 057_inlinenamespace
 9/98 Test  #7: 007_b ........................................   Passed    0.13 sec
      Start 57: 058_strong_enum
10/98 Test #11: 011_category .................................   Passed    0.13 sec
      Start 58: 059_template
11/98 Test #17: 018_def ......................................   Passed    0.12 sec
      Start 59: 060_details
12/98 Test  #8: 008_brief ....................................   Passed    0.14 sec
      Start 60: 061_Únicod€_file
13/98 Test #15: 016_copydoc ..................................   Passed    0.13 sec
      Start 61: 064_castoperator
14/98 Test #16: 017_copyright ................................   Passed    0.13 sec
      Start 62: 065_tilde
15/98 Test #19: 020_only .....................................   Passed    0.13 sec
      Start 63: 066_property_initializer
16/98 Test #14: 015_cond .....................................   Passed    0.14 sec
      Start 64: 067_link_varargs
17/98 Test #12: 013_class ....................................   Passed    0.15 sec
      Start 65: 068_ref_varargs
18/98 Test #20: 021_dontinclude ..............................   Passed    0.15 sec
      Start 66: 069_link_variadic_template
19/98 Test #22: 023_e ........................................   Passed    0.15 sec
      Start 67: 070_ref_variadic_template
20/98 Test #23: 024_if .......................................   Passed    0.15 sec
      Start 68: 071_enum_in_anon_ns
21/98 Test #21: 022_dot ......................................   Passed    0.16 sec
      Start 69: 072_using
22/98 Test #25: 026_exception ................................   Passed    0.15 sec
      Start 70: 073_typed_enum
23/98 Test #29: 030_htmlinclude ..............................   Passed    0.15 sec
      Start 71: 074_ref
24/98 Test #27: 028_formula ..................................   Passed    0.16 sec
      Start 72: 075_argmatch
25/98 Test #31: 032_include ..................................   Passed    0.15 sec
      Start 73: 076_emojis
26/98 Test #28: 029_hideinit .................................   Passed    0.16 sec
      Start 74: 077_no_xml_namespace_members_in_file_scope
27/98 Test #24: 025_example ..................................   Passed    0.17 sec
      Start 75: 078_xml_namespace_members_in_file_scope
28/98 Test #33: 034_internal .................................   Passed    0.16 sec
      Start 76: 079_tableofcontents
29/98 Test #34: 035_invariant ................................   Passed    0.16 sec
      Start 77: 080_extract_private_virtual
30/98 Test #32: 033_internal .................................   Passed    0.17 sec
      Start 78: 081_brief_lists
31/98 Test #41: 042_package ..................................   Passed    0.15 sec
      Start 79: 082_decl_def
32/98 Test #30: 031_image ....................................   Passed    0.18 sec
      Start 80: 083_decl_def
33/98 Test #37: 038_n ........................................   Passed    0.17 sec
      Start 81: 084_markdown_pre
34/98 Test #38: 039_name .....................................   Passed    0.17 sec
      Start 82: 085_tooltip
35/98 Test #39: 040_namespace ................................   Passed    0.17 sec
      Start 83: 086_style_tags
36/98 Test #35: 036_link .....................................   Passed    0.18 sec
      Start 84: 087_public_type
37/98 Test #44: 045_refitem ..................................   Passed    0.17 sec
      Start 85: 088_module_var
38/98 Test #40: 041_overload .................................   Passed    0.18 sec
      Start 86: 089_module_proc
39/98 Test #43: 044_section ..................................   Passed    0.17 sec
      Start 87: 090_interfaces
40/98 Test #48: 049_snippet ..................................   Passed    0.17 sec
      Start 88: 091_module_with_procs
41/98 Test #36: 037_msc ......................................   Passed    0.19 sec
      Start 89: 092_more_interfaces
42/98 Test #18: 019_defgroup .................................   Passed    0.23 sec
      Start 90: 093_interface_inside_routine
43/98 Test  #9: 009_bug ......................................   Passed    0.25 sec
      Start 91: 094_old_style_fortran_routines
44/98 Test #47: 048_showinit .................................   Passed    0.18 sec
      Start 92: 095_inheritance
45/98 Test #45: 046_related ..................................   Passed    0.19 sec
      Start 93: 096_namespace_inheritance
46/98 Test #42: 043_page .....................................   Passed    0.20 sec
      Start 94: 097_type_with_procedures
47/98 Test #26: 027_extends ..................................   Passed    0.23 sec
      Start 95: 098_abstract_type
48/98 Test #46: 047_return ...................................   Passed    0.19 sec
      Start 96: 099_a
49/98 Test #49: 050_verbatim .................................   Passed    0.17 sec
      Start 97: 100_a
50/98 Test #50: 051_escape ...................................   Passed    0.17 sec
      Start 98: 101_static_assert
51/98 Test #51: 052_tilde ....................................   Passed    0.17 sec
52/98 Test #52: 053_tilde ....................................   Passed    0.17 sec
53/98 Test #54: 055_markdown .................................   Passed    0.16 sec
54/98 Test #53: 054_parblock .................................   Passed    0.16 sec
55/98 Test #60: 061_Únicod€_file .........................***Failed    0.14 sec
1..1
not ok - [061_Únicod€_file.cpp]: test a file with unicode characters in the name
-------------------------------------
Non-existing file /home/tkloczko/rpmbuild/BUILD/doxygen-1.9.7/x86_64-redhat-linux-gnu/testing/test_output_061/out/061___xC3_x9Anicod_xE2_x82_xAC__file_8cpp.xml after 'check:' statement
-------------------------------------
1 out of 1 tests failed

56/98 Test #55: 056_latexinclude .............................   Passed    0.16 sec
57/98 Test #58: 059_template .................................   Passed    0.15 sec
58/98 Test #62: 065_tilde ....................................   Passed    0.15 sec
59/98 Test #56: 057_inlinenamespace ..........................   Passed    0.17 sec
60/98 Test #61: 064_castoperator .............................   Passed    0.16 sec
61/98 Test #63: 066_property_initializer .....................   Passed    0.16 sec
62/98 Test #65: 068_ref_varargs ..............................   Passed    0.15 sec
63/98 Test #64: 067_link_varargs .............................   Passed    0.16 sec
64/98 Test #57: 058_strong_enum ..............................   Passed    0.19 sec
65/98 Test #68: 071_enum_in_anon_ns ..........................   Passed    0.14 sec
66/98 Test #69: 072_using ....................................   Passed    0.14 sec
67/98 Test #72: 075_argmatch .................................   Passed    0.13 sec
68/98 Test #66: 069_link_variadic_template ...................   Passed    0.16 sec
69/98 Test #67: 070_ref_variadic_template ....................   Passed    0.16 sec
70/98 Test #71: 074_ref ......................................   Passed    0.15 sec
71/98 Test #70: 073_typed_enum ...............................   Passed    0.15 sec
72/98 Test #74: 077_no_xml_namespace_members_in_file_scope ...   Passed    0.14 sec
73/98 Test #75: 078_xml_namespace_members_in_file_scope ......   Passed    0.14 sec
74/98 Test #59: 060_details ..................................   Passed    0.21 sec
75/98 Test #77: 080_extract_private_virtual ..................   Passed    0.14 sec
76/98 Test #76: 079_tableofcontents ..........................   Passed    0.15 sec
77/98 Test #81: 084_markdown_pre .............................   Passed    0.13 sec
78/98 Test #79: 082_decl_def .................................   Passed    0.14 sec
79/98 Test #80: 083_decl_def .................................   Passed    0.14 sec
80/98 Test #78: 081_brief_lists ..............................   Passed    0.15 sec
81/98 Test #83: 086_style_tags ...............................   Passed    0.13 sec
82/98 Test #82: 085_tooltip ..................................   Passed    0.14 sec
83/98 Test #86: 089_module_proc ..............................   Passed    0.13 sec
84/98 Test #88: 091_module_with_procs ........................   Passed    0.13 sec
85/98 Test #90: 093_interface_inside_routine .................   Passed    0.12 sec
86/98 Test #85: 088_module_var ...............................   Passed    0.14 sec
87/98 Test #91: 094_old_style_fortran_routines ...............   Passed    0.13 sec
88/98 Test #84: 087_public_type ..............................   Passed    0.15 sec
89/98 Test #95: 098_abstract_type ............................   Passed    0.13 sec
90/98 Test #96: 099_a ........................................   Passed    0.12 sec
91/98 Test #94: 097_type_with_procedures .....................   Passed    0.13 sec
92/98 Test #98: 101_static_assert ............................   Passed    0.12 sec
93/98 Test #89: 092_more_interfaces ..........................   Passed    0.15 sec
94/98 Test #87: 090_interfaces ...............................   Passed    0.16 sec
95/98 Test #73: 076_emojis ...................................   Passed    0.22 sec
96/98 Test #97: 100_a ........................................   Passed    0.15 sec
97/98 Test #92: 095_inheritance ..............................   Passed    0.17 sec
98/98 Test #93: 096_namespace_inheritance ....................   Passed    0.17 sec

99% tests passed, 1 tests failed out of 98

Total Test time (real) =   0.45 sec

The following tests FAILED:
         60 - 061_Únicod€_file (Failed)
Errors while running CTest
```
</details>

**Comment by albert-github** [comment]
Would be good to have some more information, it looks a bit like a locale setting.

- what is the result of `make tests TEST_FLAGS="--id 61 --noredir` (run in the build directory)
- is there a reason for using the `-E '^(012_cite)'`

**Comment by doxygen** [comment]
Looks indeed like a locale setting, since the same tests run fine in GitHub actions. What does

```
echo $LANG && echo $LC_ALL && echo $LC_CTYPE
```

output?

**Comment by ngothan** [comment]
i tried following:

unset LANG && unset $LC_ALL && unset $LC_CTYPE

make -C redhat-linux-build/ tests  TEST_FLAGS="--id 61 --noredir"

[100%] Running doxygen tests...
cd /builddir/build/BUILD/doxygen-1.9.7/redhat-linux-build/testing && /usr/bin/python3 /builddir/build/BUILD/doxygen-1.9.7/testing/runtests.py --doxygen /builddir/build/BUILD/doxygen-1.9.7/redhat-linux-build/bin/doxygen --inputdir /builddir/build/BUILD/doxygen-1.9.7/testing --outputdir /builddir/build/BUILD/doxygen-1.9.7/redhat-linux-build/testing
1..1
not ok - [061_Únicod€_file.cpp]: test a file with unicode characters in the name
-------------------------------------
Non-existing file /builddir/build/BUILD/doxygen-1.9.7/redhat-linux-build/testing/test_output_061/out/061___xC3_x9Anicod_xE2_x82_xAC__file_8cpp.xml after 'check:' statement
-------------------------------------
1 out of 1 tests failed

The problem is that the file name of created outfile is different.  

filename of created output:
testing/test_output_061/out/061___u_xCC_x81nicod_xE2_x82_xAC__file_8cpp.xml

filename of expected output
testing/test_output_061/out/061___xC3_x9Anicod_xE2_x82_xAC__file_8cpp.xml

**Comment by albert-github** [comment]
@ngothan 

- what do you get when you initially do: `echo $LANG && echo $LC_ALL && echo $LC_CTYPE`?
- problem might also be that even when unset `make` will not pickup the unset values when starting subshells

**Comment by ngothan** [comment]
i got following output:

```
<mock-chroot> sh-5.2# echo $LANG && echo $LC_ALL && echo $LC_CTYPE
C.UTF-8


<mock-chroot> sh-5.2#
```

**Comment by kloczek** [comment]
> @ngothan
> 
> * what do you get when you initially do: `echo $LANG && echo $LC_ALL && echo $LC_CTYPE`?

I'm using rpm to build all my packages and rpm always builds with `LOCALE=C`.

> * problem might also be that even when unset `make` will not pickup the unset values when starting subshells

This test actually more tests latex than doxygem. To squize build dependencies I've sisabled it.

**Comment by albert-github** [comment]
@kloczek what is the relation here with LaTeX? the test is about the filename and the test on xml where a predefined set set of expected answers are given (in the repository) that are compared with the actual files. Due to different locales it might be that the representation of the actual file is different from the expected file.

**Comment by kloczek** [comment]
> @kloczek what is the relation here with LaTeX?

There is no relation with this issue.

> Due to different locales it might be that the representation of the actual file is different from the expected file.

As I wrote rpm buolds always with locale 'C.UTF-8'. In my devel env I';m using as well `C.UTF-8`
```console
[tkloczko@pers-jacek protobuf-23.1]$ locale
LANG=C.UTF-8
LC_CTYPE="C.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_PAPER="C.UTF-8"
LC_NAME="C.UTF-8"
LC_ADDRESS="C.UTF-8"
LC_TELEPHONE="C.UTF-8"
LC_MEASUREMENT="C.UTF-8"
LC_IDENTIFICATION="C.UTF-8"
LC_ALL=
```

**Comment by albert-github** [comment]
One of the things that was not addressed was the fact about "where did you get the sources from". Based on issue #10094 it might be that you have taken the source tar ball and we have concluded that this one wasn't correct.

- in case you downloaded the src tar ball can you download:  https://www.doxygen.nl/files/doxygen-1.9.7.repack.src.tar.gz (md5sum: c51abdaf88c3cf6b79c46f15e983725d) and test again?

**Comment by ngothan** [comment]
@albert-github i downloaded new source tar ball from above and tested. I confirmed the bug is gone with new the source tar ball 

Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by kloczek** [comment]
Just tested new version and all looks ok now.
Thank you 👍
